src/main/java/org/opensearch/neuralsearch/processor/NormalizationProcessor.java

```diff
@@ -135,4 +144,36 @@ private <Result extends SearchPhaseResult> Optional<FetchSearchResult> getFetchS
         Optional<Result> optionalFirstSearchPhaseResult = searchPhaseResults.getAtomicArray().asList().stream().findFirst();
         return optionalFirstSearchPhaseResult.map(SearchPhaseResult::fetchResult);
     }
+
+    private void recordStats(ScoreNormalizationTechnique normalizationTechnique, ScoreCombinationTechnique combinationTechnique) {
+        EventStatsManager.increment(EventStatName.NORMALIZATION_PROCESSOR_EXECUTIONS);
+        recordNormalizationCombinationTechniqueStats(normalizationTechnique, combinationTechnique);
+    }
+
+    void recordNormalizationCombinationTechniqueStats(
+        ScoreNormalizationTechnique normalizationTechnique,
+        ScoreCombinationTechnique combinationTechnique
+    ) {
+        switch (normalizationTechnique.techniqueName()) {
```

Comment:
nit: imho code is more clean if you use approach with map, where key is technique name and value is functional interface that does increment the corresponding counter, something like in factories for norm/comb techniques [ref](https://github.com/opensearch-project/neural-search/blob/main/src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombinationFactory.java)

Reply:
Sure, see my recent changes. I added a helper to unwrap the nullable safely as well.